### PR TITLE
Ignore non-primary mouse input.

### PR DESCRIPTION
**Pull request by cdata**
Addresses https://github.com/PolymerElements/paper-radio-button/issues/33 and probably others.

### CODE REVIEW

File: iron-button-state.html [hunk 1]
```diff
@@ -111,7 +113,33 @@
       this.fire('change');
     },
 
-    _downHandler: function() {
+    _eventSourceIsPrimaryInput: function(event) {
+      event = event.detail.sourceEvent || event;
+
+      // Always true for non-mouse events..
```
Comment (notwaldorf) [inline iron-button-state.html]: Heh, teensy nit: one trailing dot or three. Two is just weird :)

**Comment by notwaldorf** [comment]
This lgtm, though that buttons api makes me want to cry. Did I read it right that there's three ways to indicate a click?

**Comment by cdata** [comment]
> three ways to indicate a click

I had a long session of verbalized disbelief therapy with @azakus about this today. TL;DR `MouseEvent` be crazy (and this PR doesn't even cover all the crazy).

**Comment by azakus** [comment]
This looks like something I should maybe handle in gesture directly. Does it ever make sense to get a `down` or `up` for a right click?

**Comment by notwaldorf** [comment]
@azakus @cdata I would say not getting a `down` or `up` makes sense to me. The right click is always nommed by the browser to show the context menu, right? I don't think any elements could do anything useful with it so... 🔮

**Comment by cdata** [comment]
@azakus do you mind if we merge this in the mean time, and then take it out when your patch to gestures lands?

**Comment by azakus** [comment]
SGTM

**Comment by anthonytordillos** [comment]
Does this also fix PolymerElements/paper-button#24 (marked as a dupe), or should that issue be reopened?

**Pull request merged**